### PR TITLE
docs: fix & improve parallel algorithms documentation 3

### PR DESCRIPTION
**Pull request by bhumitattarde**

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/algorithms/adjacent_find.hpp [hunk 1]
```diff
@@ -15,8 +15,8 @@
 namespace hpx {
     /// Searches the range [first, last) for two consecutive identical elements.
```
Comment (dimitraka) [inline libs/core/algorithms/include/hpx/parallel/algorithms/adjacent_find.hpp:16]: line 48 should be:

`template <typename InIter, typename Pred = hpx::parallel::v1::detail::equal_to>`

File: libs/core/algorithms/include/hpx/parallel/algorithms/adjacent_find.hpp [hunk 2]
```diff
@@ -113,8 +113,7 @@ namespace hpx {
     ///
     template <typename ExPolicy, typename FwdIter,
         typename Pred = detail::equal_to>
```
Comment (dimitraka) [inline libs/core/algorithms/include/hpx/parallel/algorithms/adjacent_find.hpp:115]: This should be `typename Pred = hpx::parallel::v1::detail::equal_to` as well.

File: libs/core/algorithms/include/hpx/parallel/algorithms/generate.hpp [hunk 3]
```diff
@@ -56,19 +56,50 @@ namespace hpx {
...
-    /// \returns  The \a replace_if algorithm returns a \a hpx::future<FwdIter>
+    /// \returns  The \a generate algorithm returns a \a hpx::future<FwdIter>
     ///           if the execution policy is of type
     ///           \a sequenced_task_policy or
     ///           \a parallel_task_policy
     ///           and returns \a FwdIter otherwise.
-    ///           It returns \a last.
     ///
     template <typename ExPolicy, typename FwdIter, typename F>
     typename util::detail::algorithm_result<ExPolicy, FwdIter>::type
```
Comment (dimitraka) [inline libs/core/algorithms/include/hpx/parallel/algorithms/generate.hpp:66]: this should be `typename hpx::parallel::util::detail::algorithm_result<ExPolicy, FwdIter>::type`

same applies to other places as well

**Comment by hkaiser** [comment]
@bhumitattarde please rebase onto master to fix the inspect errors

**Comment by hkaiser** [comment]
@bhumitattarde any new on this?

**Comment by bhumitattarde** [comment]
@hkaiser I'm anticipating issues with clang-format so please hold on to merging this for a bit.

**Comment by hkaiser** [comment]
@bhumitattarde just a heads-up: I have fixed the clang-format issues and have rebased your branch onto master.

**Comment by bhumitattarde** [comment]
> @bhumitattarde just a heads-up: I have fixed the clang-format issues and have rebased your branch onto master.

That is very kind of you. Thank you very much! You can merge the PR if there is nothing remaining.

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**